File: java-advent-of-code/src/test/java/advent/twenty_seventeen/Day19ASeriesOfTubesTest.java
package advent.twenty_seventeen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day19ASeriesOfTubesTest {
    @Test
    public void checkReal() {
        DailyProblem<String, Integer> dp = new Day19ASeriesOfTubes("/twenty_seventeen/Day19-ASeriesOfTubes-input.txt");

        assertEquals("EPYDUXANIT", dp.getPart1Answer());
        assertEquals(17544, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<String, Integer> dp = new Day19ASeriesOfTubes("/twenty_seventeen/Day19-ASeriesOfTubes-example#1.txt");

        assertEquals("ABCDEF", dp.getPart1Answer());
        assertEquals(38, dp.getPart2Answer().intValue());
    }
}

File: scala-advent-of-code/src/test/scala/advent/twenty_seventeen/Day19ASeriesOfTubesTest.scala
package advent.twenty_seventeen

import org.scalatest.{FlatSpec, Matchers}

class Day19ASeriesOfTubesTest  extends FlatSpec with Matchers {
  "2017 Day 19 - Input" should "calculate answers" in {
    val dp = new Day19ASeriesOfTubes("/twenty_seventeen/Day19-ASeriesOfTubes-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day19ASeriesOfTubes("/twenty_seventeen/Day19-ASeriesOfTubes-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
